pkg/crypto/tinkcrypto/primitive/composite/ecdhes/ecdhes.go

```diff
@@ -0,0 +1,104 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+// Package ecdhes provides implementations of payload encryption using ECDH-ES KW key wrapping with AEAD primitives.
+//
+// The functionality of ecdhes Encryption is represented as a pair of
+// primitives (interfaces):
+//
+//  * ECDHESEncrypt for encryption of data and aad for a given list of recipients keys
+//
+//  * ECDHESDecrypt for decryption of data for a certain recipient key and returning decrypted plaintext
+//
+//
+// Example:
+//
+//  package main
+//
+//  import (
+//      "bytes"
+//
+//      "github.com/google/tink/go/keyset"
+//
+//	    ecdhessubtle "github.com/hyperledger/aries-framework-go/pkg/crypto/tinkcrypto/primitive/composite/ecdhes/subtle"
+//      "github.com/aries-framework-go/pkg/crypto/tinkcrypto/composite/ecdhes"
+//  )
+//
+//  func main() {
+//      // create recipient side keyset handle
+//      recKH, err := keyset.NewHandle(ecdhes.ECDHES256AES256GCMKeyTemplate())
+//      if err != nil {
+//          //handle error
+//      }
+//
+//      // extract recipient public keyset handle and key
+//      recPubKH, err := recKH.Public()
+//      if err != nil {
+//          //handle error
+//      }
+//
+//      buf := new(bytes.Buffer)
+//      pubKeyWriter := ecdhes.NewWriter(buf)
+//      err = recPubKH.WriteWithNoSecrets(pubKeyWriter)
+//      if err != nil {
+//          //handle error
+//      }
+//
+//      ecPubKey := new(ecdhessubtle.ECPublicKey)
+//      err := json.Unmarshal(buf.Bytes(), ecPubKey)
+//
+//      // now create sender keyset handle with recipient public key (ecPubKey)
+//      sKH, err := keyset.NewHandle(ECDHES256AES256GCMKeyTemplateWithRecipients([]ecdhessubtle.ECPublicKey{*ecPubKey}))
+//      if err != nil {
+//          // handle error
+//      }
+//
+//      // for more recipient keys pass in a list: []ecdhessubtle.ECPublicKey{*ecPubKey1, *ecPubKey2, *ecPubKey3, etc.})
+//      // at least 1 recipient is required.
+//
+//      // extract sender public keyset handle to encrypt
+//      senderPubKH, err := sKH.Public()
+//      if err != nil {
+//          //handle error
+//      }
+//
+//      e := ecdhes.NewECDHESEncrypt(senderPubKH)
+//
+//      ct, err = e.Encrypt([]byte("secret message"), []byte("some aad"))
+//      if err != nil {
+//          // handle error
+//      }
+//
+//      // get a handle on the decryption key material for a recipient
+//      // this is usually reloading the recipient's keyset handle (ie: `recKH` above) from a kms
+//      refRecKH , err := keyset.NewHandle( .....reference/rebuild `recKH` here...);
+//      d := ecdhes.NewECDHESDecrypt(refRecKH)
+//
+//      pt, err := d.Decrypt(ct)
+//      if err != nil {
+//          // handle error
+//      }
+//  }
+package ecdhes
+
+import (
+	"fmt"
+
+	"github.com/google/tink/go/core/registry"
+)
+
+// nolint: gochecknoinits
+func init() {
+	err := registry.RegisterKeyManager(newECDHESPrivateKeyManager())
```

Comment:
TODO: avoid the tink registry singleton.

Reply:
This can only be done if Tink removes their singletons. The new keys templates created in this patch are 100% Tink supported and must follow their pattern. 

Another approach would be to call these registery functions in Aries.New() using a once.Do(), let me know if it makes sense and I'll change this code.

Reply:
I've been thinking of the other approach, I don't think we can move these calls to Aries.New() as all unit tests will fail unless we explicitly call Aries.New() in all the tests requiring Tink keys.

I think for now, we will have to keep this code as is until Tink changes their singleton pattern.

Reply:
Added TODO